catalyst_voices/Earthfile

```
@@ -24,6 +29,34 @@ src:
     FROM +deps
     COPY --dir pubspec.yaml lib packages web test test_driver integration_test .
 
+# code-generator - Generates flutter code.
+# Based on Catalyst Gateway OpenAPI specifications generates models, clients
+# and serialization logic.
+code-generator:
+    FROM +deps
+    COPY --dir pubspec.yaml lib packages web test test_driver integration_test catalyst_voices
+    COPY ../+repo-catalyst-voices-packages/repo .
+    WORKDIR /frontend/catalyst_voices/packages/catalyst_voices_services
+    COPY ../catalyst-gateway+build/doc/cat-gateway-api.json openapi/cat-gateway-api.json
+    RUN flutter pub get
+    RUN dart run build_runner build --delete-conflicting-outputs
+    SAVE ARTIFACT lib/generated
```

Maybe you can pass an `ARG save_locally` to this target and when it is set to true it saves the generated code locally? so you can remove the `code-generator-local` target? Something like:
```suggestion
IF [ "$save_locally" = "true" ]
        SAVE ARTIFACT  lib/generated AS LOCAL  lib/generated
    ELSE
        SAVE ARTIFACT  lib/generated
    END
```

---

Yeah that's a good idea!